src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/SdkContentTests.cs

```diff
@@ -24,16 +32,171 @@ public class SdkContentTests : SmokeTests
+            // we need to replace the version placeholders with wildcards and find the path through path matching.
+            string file = Path.Combine(msftSdkPath, relativePath);
+            Matcher matcher = BaselineHelper.GetFileMatcherFromPath(relativePath);
+
+            file = FindMatchingFilePath(msftSdkPath, matcher, relativePath);
+
+            if (!File.Exists(file))
+            {
+                continue;
+            }
+
+            AssemblyName assemblyName = AssemblyName.GetAssemblyName(file);
+            msftSdkAssemblyVersions.Add(BaselineHelper.RemoveVersions(relativePath), GetVersion(assemblyName));
+        }
+        return msftSdkAssemblyVersions;
+    }
+
+    // It's known that assembly versions can be different between builds in their revision field. Disregard that difference
+    // by excluding that field in the output.
+    private static Version? GetVersion(AssemblyName assemblyName)
+    {
+        if (assemblyName.Version is not null)
+        {
+            return new Version(assemblyName.Version.ToString(3));
+        }
+
+        return null;
+    }
+
+    private string FindMatchingFilePath(string rootDir, Matcher matcher, string representativeFile)
+    {
+        foreach (string file in Directory.EnumerateFiles(rootDir, "*", SearchOption.AllDirectories))
+        {
+            if (matcher.Match(rootDir, file).HasMatches)
+            {
+                return file;
+            }
+        }
+
+        Assert.Fail($"Unable to find matching file for '{representativeFile}' in '{rootDir}'.");
+        return string.Empty;
+    }
+
+    private Dictionary<string, Version?> GetSbSdkAssemblyVersions(string sbSdkPath)
+    {
+        IEnumerable<string> exclusionFilters = GetSdkDiffExclusionFilters(SourceBuildSdkType)
+            .Select(filter => filter.TrimStart("./".ToCharArray()));
+        Dictionary<string, Version?> sbSdkAssemblyVersions = new();
+        foreach (string file in Directory.EnumerateFiles(sbSdkPath, "*", SearchOption.AllDirectories))
+        {
+            string fileExt = Path.GetExtension(file);
+            if (fileExt.Equals(".dll", StringComparison.OrdinalIgnoreCase) ||
+                fileExt.Equals(".exe", StringComparison.OrdinalIgnoreCase))
+            {
+                try
+                {
+                    AssemblyName assemblyName = AssemblyName.GetAssemblyName(file);
+                    string relativePath = Path.GetRelativePath(sbSdkPath, file);
+                    string normalizedPath = BaselineHelper.RemoveVersions(relativePath);
+
+                    if (!IsFileExcluded(normalizedPath, exclusionFilters))
+                    {
+                        sbSdkAssemblyVersions.Add(normalizedPath, GetVersion(assemblyName));
+                    }
+                }
+                catch (BadImageFormatException)
+                {
+                    // Ignore non-.NET files
```

Comment:
Curious how many of these there are and what they are?

Reply:
Zero, it seems. This was defensive programming. After thinking about it, perhaps it's better to just remove this and have it fail so we can be aware of any situation where this would fail. I've removed the try..catch.